modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -1898,6 +1898,55 @@ def test___setitem__unhashable_list():
     df_equals(modin_df, pandas_df)
 
 
+def test_setitem_unhashable_key():
+    data = test_data["float_nan_data"]
+    source_modin_df, source_pandas_df = pd.DataFrame(data), pandas.DataFrame(data)
+    count_rows = source_modin_df.shape[0]
+
+    def _make_copy(df1, df2):
+        return df1.copy(deep=True), df2.copy(deep=True)
+
+    for key in (["col1", "col2"], ["new_col1", "new_col2"]):
+        # 1d list case
+        value = [1, 2]
+        modin_df, pandas_df = _make_copy(source_modin_df, source_pandas_df)
```

Comment:
Why do we need to make a copy?

Reply:
`setitem` operation is performed inplace so that the effects do not overlap each other we can make copies.